freshmaker/container.py

```diff
@@ -242,6 +244,80 @@ def resolve(self, pyxis_instance: PyxisGQL, koji_session: KojiService, children=
         self.resolve_build_metadata(koji_session)
         self.resolve_compose_sources()
 
+    def resolve_content_sets(
+        self, pyxis_instance: PyxisGQL, koji_session: KojiService, children=None
+    ):
+        if not self.content_sets_by_arch:
+            log.info(
+                "Container image %s uses following content sets: %r",
+                self.nvr,
+                self.content_sets_by_arch,
+            )
+            if self.content_sets_source is None:
+                self.content_sets_source = "pyxis_container_image"
+            return
+        for child in children:
+            if not child.content_sets_by_arch:
+                child.resolve(pyxis_instance, koji_session, child)
+            if not child.content_sets_by_arch:
+                continue
+
+            log.info(
+                "Container image %s does not have 'content-sets' set "
+                "in Lightblue. Using child image %s content_sets: %r",
+                self.nvr,
+                child.nvr,
+                child.content_sets_by_arch,
+            )
+            self.content_sets_by_arch = child.content_sets_by_arch
+            return
+
+        log.warning(
+            "Container image %s does not have 'content_sets' set "
+            "in Lightblue as well as its children, this "
+            "is suspicious.",
+            self.nvr,
+        )
+
+    def resolve_published(self, pyxis_instance: PyxisGQL):
+        # Get the published version of this image to find out if the image
+        # was actually published.
+        images = pyxis_instance.find_images_by_nvrs(self.nvr, include_rpms=True)
+        if images:
+            self.published = True
+        else:
+            self.published = False
+
+            # Usually we do not store complete RPM manifest, but when
+            # image is unpublished, we need complete RPM manifest in order
+            # to check for possible unpublished RPMs.
+            # We do not want to get the complete manifest for every container
+            # image, because it is relatively big, so fetch it only when needed.
+            images = pyxis_instance.find_images_by_nvrs(self.nvr, include_rpms=True)
+            if images:
+                self.rpms = self.convert_rpms(
+                    images[0]["edges"]["rpm_manifest"]["data"]["rpms"]
+                )
```

Comment:
This is weird, you set `self.rpms` in `convert_rpms` and then assign `None` to `self.rpms`, also you repeat the code in `load` and `convert_rpms`.